src/herder/TxSetFrame.cpp

```diff
@@ -308,9 +330,20 @@ TxSetFrame::makeFromTransactions(TxPhases const& txPhases, Application& app,
         }
 
         auto& invalid = invalidTxs[i];
-        validatedPhases.emplace_back(
-            TxSetUtils::trimInvalid(txs, app, lowerBoundCloseTimeOffset,
-                                    upperBoundCloseTimeOffset, invalid));
+#ifdef BUILD_TESTS
+        if (!skipValidation)
+        {
+#endif
+            validatedPhases.emplace_back(
+                TxSetUtils::trimInvalid(txs, app, lowerBoundCloseTimeOffset,
+                                        upperBoundCloseTimeOffset, invalid));
+#ifdef BUILD_TESTS
+        }
+        else
+        {
+            validatedPhases.emplace_back(txs);
+        }
+#endif
```

Comment:
nit: a slightly more readable way would be
```
#ifdef BUILD_TESTS
        if (skipValidation)
        {
             // Add unvalidated
        }
       else
#endif
        {
             // Add validated
        }
```

Reply:
Updated. I still wrapped the scope in macros just in case (I don't quite like having some arbitrary scopes in production build)